auth/auth.go

```diff
@@ -74,44 +105,56 @@ func (r *RenewingChecker) IsVPNExpired() (bool, error) {
 	}
 
 	data := cfg.TokensData[cfg.AutoConnectData.ID]
-	if isTokenExpired(data.ServiceExpiry) {
-		if err := r.updateVpnExpirationDate(&data); err != nil {
+	if r.expChecker.isExpired(data.ServiceExpiry) {
+		if err := r.fetchSaveServices(cfg.AutoConnectData.ID, &data); err != nil {
 			return true, fmt.Errorf("updating service expiry token: %w", err)
 		}
 		if err := r.cm.SaveWith(saveVpnExpirationDate(cfg.AutoConnectData.ID, data)); err != nil {
 			return true, fmt.Errorf("saving config: %w", err)
 		}
 	}
 
-	return isTokenExpired(data.ServiceExpiry), nil
+	return r.expChecker.isExpired(data.ServiceExpiry), nil
 }
 
-// IsDedicatedIPExpired is used to check whether the user is allowed to use dedicated IP servers
-func (r *RenewingChecker) IsDedicatedIPExpired() (bool, error) {
+func (r *RenewingChecker) GetDedicatedIPServices() ([]DedicatedIPService, error) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	var cfg config.Config
-	if err := r.cm.Load(&cfg); err != nil {
-		return true, fmt.Errorf("loading config: %w", err)
+	services, err := r.fetchServices()
+	if err != nil {
+		return nil, fmt.Errorf("fetching available services: %w", err)
 	}
 
-	data := cfg.TokensData[cfg.AutoConnectData.ID]
-	if isTokenExpired(data.DedicatedIPExpiry) {
-		if err := r.updateVpnExpirationDate(&data); err != nil {
-			return true, fmt.Errorf("updating service expiry token: %w", err)
-		}
-		if err := r.cm.SaveWith(saveVpnExpirationDate(cfg.AutoConnectData.ID, data)); err != nil {
-			return true, fmt.Errorf("saving config: %w", err)
+	noDIPServers := true
+	noDIPServices := true
+	dipServices := []DedicatedIPService{}
+	for _, service := range services {
+		if service.Service.ID == DedicatedIPServiceID && !r.expChecker.isExpired(service.ExpiresAt) {
+			noDIPServices = false
+			serversLen := len(service.Details.Servers)
+			if serversLen != 1 {
+				log.Println(internal.ErrorPrefix,
+					"unexpected number of dedicated ip servers in service, expected 1, is", serversLen)
+				continue
+			}
+			noDIPServers = false
+			dipServices = append(dipServices,
```

Comment:
Maybe we should exit on first match? what is the point of having multiple DIP services? it's only data structure is like that, but in Nordaccount.com it is not possible to buy multiple services, afaik.

Reply:
It is possible to have multiple DIP services - confirmed by API team.